core/src/main/java/org/apache/iceberg/PartitionData.java

```diff
@@ -72,6 +72,17 @@ public PartitionData(Types.StructType partitionType) {
     this.stringSchema = schema.toString();
   }
 
+  public PartitionData() {
+    // This construtor has niche case of being used at time for rest serial/deserilization
+    // where we do not have the partition spec and can not populate these values.
+    // these values will be refreshed before returning to client engine
+    this.partitionType = null;
+    this.size = 0;
+    this.data = new Object[size];
+    this.schema = null;
+    this.stringSchema = null;
+  }
```

Comment:
Discussed with @rahil-c technically we could avoid this constructor by passing in a null value to the GenericDataFile and rebinding that later, but the bigger question is if we want the server to just send back a serialized spec

Reply:
Will fix this in next revision.